lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -843,6 +875,105 @@ void FunctionEntrypointAnalyzer::interproceduralPropagation() {
   }
 }
 
+void FEA::initializeMapForDeductions(FunctionSummary &Summary,
+                                     abi::RegisterState::Map &Map) {
+  using namespace abi;
+  using namespace ABIAnalyses;
+  using RegisterState = abi::RegisterState::Values;
+  using Register = model::Register::Values;
+
+  auto GPRArguments = listGeneralPurposeArgumentRegisters(Binary->DefaultABI);
+  auto GPRRetVals = listGeneralPurposeReturnValueRegisters(Binary->DefaultABI);
+  auto VRArguments = listVectorArgumentRegisters(Binary->DefaultABI);
+  auto VRRetVals = listVectorReturnValueRegisters(Binary->DefaultABI);
+
+  auto TryGetRegisterState =
+    [this](Register Reg,
+           RegisterStateMap &ABIRegisterMap) -> std::optional<RegisterState> {
+    llvm::StringRef Name = model::Register::getCSVName(Reg);
+    if (llvm::GlobalVariable *CSV = M.getGlobalVariable(Name, true)) {
+      auto It = ABIRegisterMap.find(CSV);
+      if (It != ABIRegisterMap.end())
+        if (isYesOrDead(It->second))
+          return It->second;
+    }
+
+    return {};
```

Comment:
I feel like default-constructing an `std::optional` looks a bit misleading, `std::nullopt` is also a lot more explicit about what's actually going on. Don't you agree, @antoniofrighetto?

Reply:
Yeah, I definitely agree, it's way more explicit and quickly understandable.